client/state/posts/selectors.js

```diff
@@ -24,6 +25,28 @@ export function getPost( state, globalId ) {
 }
 
 /**
+ * Returns a mapping of site ID, post ID pairing to global post ID.
+ *
+ * @param state
+ */
+export const getPostsBySiteIdAndPostId = createSelector(
+	( state ) => {
```

Comment:
Bordering on a bit too clever, but I tried rewriting this with a few more Lodash helpers:

```jsx
export const getPostsBySiteIdAndPostId = createSelector(
	( state ) => {
		return reduce( state.posts.items, ( result, post, globalId ) => {
			return setWith( result, [ post.site_ID, post.ID ], globalId, Object );
		}, {} );
	},
	( state ) => [ state.posts.items ]
);
```